### PR TITLE
[SYCLomatic] Disable some helper function code when linking open source MKL

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

File: clang/runtime/dpct-rt/include/blas_utils.hpp.inc [hunk 1]
```diff
@@ -388,6 +413,9 @@ inline void gemm_batch_impl(sycl::queue &q, oneapi::mkl::transpose a_trans,
     cgh.depends_on(e);
     cgh.host_task([=] { std::free(matrix_info); });
   });
+#else
+throw std::runtime_error("open source mkl library not support");
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/blas_utils.hpp.inc]:
```suggestion
  throw std::runtime_error("open source mkl library not support");
```

File: clang/runtime/dpct-rt/include/lapack_utils.hpp.inc [hunk 2]
```diff
@@ -886,9 +890,13 @@ inline int getrf(sycl::queue &q, std::int64_t m, std::int64_t n,
   std::size_t device_ws_size_in_element_number =
       detail::byte_to_element_number(device_ws_size, a_type);
   if (ipiv == nullptr) {
+#ifdef __INTEL_MKL__
     return detail::lapack_shim<detail::getrfnp_impl>(
         q, a_type, info, "getrfnp_batch", q, m, n, a_type, a, lda, ipiv,
         device_ws, device_ws_size_in_element_number, info);
+#else
+throw std::runtime_error("open source mkl library not support");
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/lapack_utils.hpp.inc]:
```suggestion
  throw std::runtime_error("open source mkl library not support");
```

File: clang/runtime/dpct-rt/include/rng_utils.hpp.inc [hunk 3]
```diff
@@ -419,8 +427,12 @@ public:
   inline void generate_uniform_bits(unsigned long long *output,
                                     std::int64_t n) {
     static_assert(sizeof(unsigned long long) == sizeof(std::uint64_t));
+#ifdef __INTEL_MKL__
     generate<oneapi::mkl::rng::uniform_bits<std::uint64_t>>(
         (std::uint64_t *)output, n);
+#else
+throw std::runtime_error("open source mkl library not support");
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/rng_utils.hpp.inc]:
```suggestion
  throw std::runtime_error("open source mkl library not support");
```

File: clang/runtime/dpct-rt/include/sparse_utils.hpp.inc [hunk 4]
```diff
@@ -236,12 +241,16 @@ void csrmm(sycl::queue &queue, oneapi::mkl::transpose trans, int sparse_rows,
     cgh.depends_on(e);
     cgh.host_task([=] { delete sparse_matrix_handle; });
   });
+#else
+throw std::runtime_error("open source mkl library not support");
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/sparse_utils.hpp.inc]:
```suggestion
  throw std::runtime_error("open source mkl library not support");
```

File: clang/runtime/dpct-rt/include/blas_utils.hpp.inc [hunk 5]
```diff
@@ -156,8 +156,12 @@ inline void nrm2_impl(sycl::queue &q, int n, const void *x, int incx,
   oneapi::mkl::blas::column_major::nrm2(q, n, x_buffer, incx, r_buffer);
 #else
   working_memory<Tr> res_mem(reinterpret_cast<Tr *>(result), q);
+#ifdef __INTEL_MKL__
   oneapi::mkl::blas::column_major::nrm2(q, n, reinterpret_cast<const Tx *>(x),
                                         incx, res_mem.get_ptr());
+#else
+  throw std::runtime_error("open source mkl library not support");
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/blas_utils.hpp.inc]:
```suggestion
  throw std::runtime_error("The oneAPI Math Kernel Library (oneMKL) Interfaces Project does not support this API.");
```

**Review by wenhuiNi — APPROVED** [review]

**Review by zhiweij1 — APPROVED** [review]

File: clang/runtime/dpct-rt/include/sparse_utils.hpp.inc [hunk 6]
```diff
@@ -277,17 +287,17 @@ private:
...
 /// \param [in] queue The queue where the routine should be executed. It must
 /// have the in_order property when using the USM mode.
 /// \param [in] trans The operation applied to the sparse matrix.
 /// \param [in] row_col Number of rows of the sparse matrix.
 /// \param [in] info Matrix info of the sparse matrix.
-/// \param [in] val An array containing the non-zero elements of the sparse matrix.
-/// \param [in] row_ptr An array of length \p num_rows + 1.
-/// \param [in] col_ind An array containing the column indices in index-based
-/// numbering.
+/// \param [in] val An array containing the non-zero elements of the sparse
```
Comment (tangjj11) [inline clang/runtime/dpct-rt/include/sparse_utils.hpp.inc]: should not format.

**Review by ziranzha — APPROVED** [review]

**Review by zhiweij1 — APPROVED** [review]

**Review by ShengchenJ — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**